timemachine/cpp/src/nonbonded_interaction_group.hpp

```diff
@@ -69,13 +71,14 @@ template <typename RealType> class NonbondedInteractionGroup : public Potential
     void set_atom_idxs_device(
         const int NC, const int NR, unsigned int *d_column_idxs, unsigned int *d_row_idxs, const cudaStream_t stream);
 
-    void set_atom_idxs(const std::vector<int> &atom_idxs);
+    void set_atom_idxs(const std::vector<int> &row_atom_idxs, const std::optional<std::vector<int>> &col_atom_idxs);
 
     NonbondedInteractionGroup(
         const int N,
         const std::vector<int> &row_atom_idxs,
         const double beta,
         const double cutoff,
+        const std::optional<std::set<int>> &col_atom_idxs,
```

Comment:
I really don't want an `optional` for something this low-level, I'd rather the signature just be:
``` c++
NonbondedInteractionGroup(
        const int N,
        const std::vector<int> &row_atom_idxs,
        const std::vector<int> &col_atom_idxs,
        const double beta,
        const double cutoff
  )```